src/main/scala/is/hail/io/LoadMatrix.scala

```diff
@@ -20,34 +20,23 @@ class LoadMatrixParser(rvb: RegionValueBuilder, fieldTypes: Array[Type], entryTy
 
   val sep = '\t'
   val nFields: Int = fieldTypes.length
-  val cellf: (String, Long, Int, Int) => Int = entryType.fieldType(0) match {
-    case TInt32(_) => getInt
-    case TInt64(_) => getLong
-    case TFloat32(_) => getFloat
-    case TFloat64(_) => getDouble
-    case TString(_) => getString
-  }
-
+  val cellf: (String, Long, Int, Int) => Int = addType(entryType.fieldType(0))
 
   def parseLine(line: String, rowNum: Long): Unit = {
     var ii = 0
     var off = 0
     while (ii < fieldTypes.length) {
-      off = fieldTypes(ii) match {
-        case TInt32(_) => getInt(line, rowNum, ii, off)
-        case TInt64(_) => getLong(line, rowNum, ii, off)
-        case TFloat32(_) => getFloat(line, rowNum, ii, off)
-        case TFloat64(_) => getDouble(line, rowNum, ii, off)
-        case TString(_) => getString(line, rowNum, ii, off)
-      }
+      off = addType(fieldTypes(ii))(line, rowNum, ii, off)
+      ii += 1
       if (off > line.length) {
         fatal(
           s"""Error parsing row fields in row $rowNum:
              |    expected $nFields fields but only $ii found.
-             |    in file $file""".stripMargin
+             |    File: $file
+             |    Line:
+             |        ${ line.take(76) }...""".stripMargin
         )
       }
```

Comment:
I love the ellipsis, but it will be a bit confusing if there's an ellipsis when the string is small. It's kind of hard to find but we have `is.hail.utils.Truncatable`. There are implicit conversions in `is.hail.utils._` that let you call `s.truncate(76)` on any `String` `s`.